tools/crm_resource_runtime.c

```diff
@@ -1713,3 +1752,87 @@ cli_resource_move(const char *rsc_id, const char *host_name, cib_t * cib, pe_wor
 
     return rc;
 }
+
+static void
+cli_resource_why_without_rsc_and_host(cib_t *cib_conn,GListPtr resources)
+{
+    GListPtr lpc = NULL;
+    GListPtr hosts = NULL;
+
+    for (lpc = resources; lpc != NULL; lpc = lpc->next) {
+        resource_t *rsc = (resource_t *) lpc->data;
+        rsc->fns->location(rsc, &hosts, TRUE);
+	if ( hosts == NULL ) {
+	    printf("Resource %s is not running",rsc->id);
+            cli_resource_check(cib_conn, rsc);
+	} else {
+	    printf("Resource %s is running",rsc->id);
+	}
+     }
+
+     g_list_free(hosts);
+     hosts = NULL;
+}
+
+static void
+cli_resource_why_with_rsc_and_host(cib_t *cib_conn,GListPtr resources,const char* rsc_id,const char* host_uname)
+{
+    resource_t *rsc = NULL;
+
+    rsc = pe_find_resource(resources, rsc_id);
+    if(!(resource_is_running_on(rsc,host_uname))) {
+        cli_resource_check(cib_conn, rsc);
+    } else {
+        printf("Resource %s is not running on host %s",rsc->id,host_uname);
+    }
+}
+
+static void 
+cli_resource_why_without_rsc_with_host(cib_t *cib_conn,GListPtr resources,const char* host_uname)
+{
+    GListPtr allResources = get_all_resources(host_uname,resources);
```

Comment:
Nice ! I did not know that we could get it from the node_t. :slightly_smiling_face: 
This new apporach looks much cleaner and also removes the redudant code.